Tests/SourceControlTests/GitRepositoryProviderTests.swift

```diff
@@ -40,4 +40,76 @@ class GitRepositoryProviderTests: XCTestCase {
             XCTAssertThrowsError(try provider.isValidDirectory(notGitChildPath))
         }
     }
+
+    func testIsValidDirectoryThrowsPrintableError() throws {
+        try testWithTemporaryDirectory { temp in
+            let provider = GitRepositoryProvider()
+            let expectedErrorMessage = "not a git repository"
+            do {
+                try _ = provider.isValidDirectory(temp)
```

Comment:
don't you need to either use `XCTAssertThorwsError` or an `XCTFail` after this line? It's wrong if it doesn't throw but at the moment, the test will pass, right?

Reply:
Thank you, you're absolutely right. It is `XCTAssertThrowsError` now in [8840c58](https://github.com/swiftlang/swift-package-manager/pull/8152/commits/8840c58ea6d14eb153ee01220e22c4763e2ac183)